config_hub/finetune/llama-2-7b/full.yaml

```diff
@@ -1,39 +1,96 @@
+
+# The path to the base model checkpoint directory to load for finetuning. (type: <class 'Path'>, default: checkpoints/stabilityai/stablelm-base-alpha-3b)
+checkpoint_dir: checkpoints/meta-llama/Llama-2-7b-hf
+
+# Where to save checkpoints and logs. (type: <class 'Path'>, default: out/finetune/full)
+out_dir: out/finetune/full-llama2-7b
+
+# The precision to use for finetuning. Possible choices: "bf16-true", "bf16-mixed", "32-true". (type: Optional[str], default: null)
 precision: bf16-true
+
+# How many devices/GPUs to use (type: Union[int, str], default: 1)
 devices: 4
+
+# Path to a checkpoint directory to resume from in case training got interrupted. Or ``True`` to resume
+# from the latest checkpoint in ``out_dir``. (type: Union[bool, Path], default: False)
 resume: false
-seed: 1337
+
+# Data related arguments. If not provided, the default is ``litgpt.data.Alpaca``.
 data:
   class_path: litgpt.data.AlpacaGPT4
   init_args:
     mask_prompt: false
     val_split_fraction: 0.03847
-    prompt_style: "alpaca"
+    prompt_style: alpaca
     ignore_index: -100
     seed: 42
     num_workers: 4
-    download_dir: data/alpacagpt4
-    file_url: https://raw.githubusercontent.com/Instruction-Tuning-with-GPT-4/GPT-4-LLM/main/data/alpaca_gpt4_data.json
-    file_name: alpacagpt4_data_cleaned_archive.json
-checkpoint_dir: checkpoints/meta-llama/Llama-2-7b-hf
-out_dir: out/finetune/full-llama2-7b
+
+# Training related arguments. See ``litgpt.args.TrainArgs`` for details
 train:
+
+  # Number of optimizer steps between checkpoints (type: Optional[int], default: 1000)
   save_interval: 200
+
+  # Number of iterations between logging calls (type: int, default: 1)
   log_interval: 1
+
+  # Number of samples between optimizer steps across data-parallel ranks (type: int, default: 64)
   global_batch_size: 64
+
+  # Number of samples per data-parallel rank (type: int, default: 1)
   micro_batch_size: 4
+
+  # Number of iterations with learning rate warmup active (type: int, default: 100)
   lr_warmup_steps: 25
+
+  # Number of epochs to run (type: Optional[int], default: 5)
   epochs: 1
-  max_tokens: null
-  max_steps: null
+
+  # Total number of tokens to train on (type: Optional[int], default: null)
+  max_tokens:
+
+  # Limits the number of optimizer steps to run. (type: Optional[int], default: null)
```

Comment:
```suggestion
  # Limits the number of optimizer steps to run. If not provided, the number of steps depends on the dataset size, batch size, and number of epochs. (type: Optional[int], default: null)
```

Reply:
I'll keep this for future work for now. Because all the descriptions here are generally very brief. More details could be added to all of them, or not, depending on how we want to document our "Trainer".

Reply:
Yeah, it's a bit long. I just wanted to make clear that this is an argument that overrides other settings.